app/src/main/java/com/davidtomas/taskyapp/features/agenda/presentation/AgendaDetail/AgendaDetailRoot.kt

```diff
@@ -45,6 +51,13 @@ fun AgendaDetailRoot(
                 }
             }
         }
+    }
+    LaunchedEffect(key1 = isPhotoDeleted) {
+        isPhotoDeleted?.value?.let { isPhotoDeleted ->
+            if (isPhotoDeleted) agendaDetailViewModel.populateData()
+        }
+    }
+    LaunchedEffect(key1 = true) {
         agendaDetailViewModel.uiEvent.collect { event ->
             when (event) {
                 is AgendaDetailUiEvent.NavigateUp -> {
```

Comment:
This also doesn't collect these events correctly. You need something like this:
```
@Composable
private fun <T> ObserveAsEvents(flow: Flow<T>, onEvent: (T) -> Unit) {
    val lifecycleOwner = LocalLifecycleOwner.current
    LaunchedEffect(flow, lifecycleOwner.lifecycle) {
        lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
            withContext(Dispatchers.Main.immediate) {
                flow.collect(onEvent)
            }
        }
    }
}
```
For more info, I talk about the problem in this video: https://youtu.be/njchj9d_Lf8?si=0to4KwRHzEIECID8